ord_schema/orm/README.md

```diff
@@ -1,9 +1,163 @@
+storage and retrieval, they are not optimized for searching. To support field-level queries, we use the 
+[SQLAlchemy ORM](https://docs.sqlalchemy.org/en/14/orm/quickstart.html) to convert protocol buffers to entries in a 
+relational database.
+
+## Overview
+
+Conceptually, an ORM is an abstraction on top of a relational database that allows data to be manipulated using
+object-oriented programming techniques. In our case, every protocol buffer message has an associated "mapper" that wraps
+a table in a relational database. For example, here is the definition of the `Mass` message in the protocol buffer
+schema, which is used as a subfield in the `Amount` message:
+
+```protobuf
+message Mass {
+  enum MassUnit {
+    UNSPECIFIED = 0;
+    KILOGRAM = 1;
+    GRAM = 2;
+    MILLIGRAM = 3;
+    MICROGRAM = 4;
+  }
+  optional float value = 1;
+  // Precision of the measurement (with the same units as `value`).
+  optional float precision = 2;
+  MassUnit units = 3;
+}
+
+message Amount {
+  oneof kind {
+    Mass mass = 1;
+    Moles moles = 2;
+    Volume volume = 3;
+    UnmeasuredAmount unmeasured = 5;
+  }
+  // Whether the volume measurement refers to the pure substance or to the
+  // total volume of the reaction input. An example of when this field should
+  // be TRUE is when stock solutions are prepared by adding solvent to a
+  // volumetric flask already containing solute(s).
+  optional bool volume_includes_solutes = 4;
+}
+```
+
+The `Mass` message has three fields: `value` (float), `precision` (float), and `units` (enum). The corresponding ORM
+mapper looks like this:
+
+```python
+class Mass(Base):
+    # Some attributes are defined by `Base`:
+    # __tablename__ = "mass"
+    # id = Column(Integer, primary_key=True)
+    amount_id = Column(Integer, ForeignKey("amount.id", ondelete="CASCADE"), nullable=False, unique=True)
+
+    value = Column(Float)
+    precision = Column(Float)
+    units = Column(Enum(*reaction_pb2.Mass.MassUnit.keys(), name="Mass.MassUnit"))
+```
+
+The mapper defines a relational database table (`mass`) with five columns: `id` (a unique row ID); `amount_id`
+(a reference to the parent `Amount` message in the relational database); and the expected `value`, `precision`, and
+`units` fields from the protocol buffer message. Instances of the mapper correspond to individual rows in the associated
+table.
+
+## Usage
+
+### Install PostgreSQL and the RDKit extension
+
+We use PostgreSQL with the RDKit extension to support structure searches. There are a couple convenient installation
+methods (this list is not an endorsement of any particular provider):
+
+* Conda: [rdkit-postgresql](https://www.rdkit.org/docs/Install.html#installing-and-using-postgresql-and-the-rdkit-postgresql-cartridge-from-a-conda-environment)
+  
+  ```shell
+  $ conda install -c rdkit rdkit-postgresql
+  $ export PGDATA="${HOME}/rdkit-postgresql"
+  $ initdb

```

Comment:
initdb -U <username> is likely the better command to run to make the downstream 'scott':'tiger' explicit.

Reply:
done; also updated the connection string example